File: apple_division/apple_division.py
from sys import stdin

def main():
    n = int(input())
    weights = list(map(int, stdin.readline().split()))
    ans = [10**9]
    def dfs(i, running):
        if i >= n:
            ans[0] = min(abs(running), ans[0])
            return
        dfs(i+1, running + weights[i])
        dfs(i+1, running - weights[i])
    dfs(0,0)
    print(ans[0])


main()

File: apple_division/apple_division.cc
#include <iostream>
#include <vector>
#include <math.h>

using namespace std;

void dfs(int i, long running, long& min, int n, vector<int> v){
  if (i >= n) {
    min = std::min(min, abs(running));
  } else {
    dfs(i+1, running + v[i], min, n, v);
    dfs(i+1, running - v[i], min, n, v);
  }
}

int main() {
  int n;
  cin >> n;
  vector<int> weights;
  for (int i = 0; i < n; ++i) {
    int tmp;
    cin >> tmp;
    weights.push_back(tmp);
  }
  long min = pow(10, 9) + 1;
  dfs(0, 0, min, n, weights);
  cout << min << endl;
}